morpheus/stages/input/rss_source_stage.py

```diff
@@ -0,0 +1,119 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import feedparser
+import cudf
+from morpheus.utils.controllers.rss_controller import RSSController
+import mrc
+from morpheus.cli import register_stage
+import time
+import feedparser
+import pandas as pd
+from datetime import datetime, timedelta
+from morpheus.config import Config
+from morpheus.config import PipelineModes
+from morpheus.messages import MessageMeta
+from morpheus.pipeline.preallocator_mixin import PreallocatorMixin
+from morpheus.pipeline.single_output_source import SingleOutputSource
+from morpheus.pipeline.stream_pair import StreamPair
+
+
+logger = logging.getLogger(__name__)
+
+@register_stage("from-rss", modes=[PipelineModes.AE, PipelineModes.FIL, PipelineModes.NLP, PipelineModes.OTHER])
+class RSSSourceStage(PreallocatorMixin, SingleOutputSource):
+    """
+    Load RSS feed items into a pandas DataFrame.
+
+    Parameters
+    ----------
+    c : morpheus.config.Config
+        Pipeline configuration instance.
+    feed_input : str
+        The URL of the RSS feed.
+    interval_secs : float, optional, default = 600
+        Interval in seconds between fetching new feed items.
+    stop_after: int, default = 0
+        Stops ingesting after emitting `stop_after` records (rows in the dataframe). Useful for testing. Disabled if `0`
+    max_retries : int, optional, default = 3
+        Maximum number of retries for fetching entries on exception.
+    """
+    def __init__(self, c: Config, feed_input: str, interval_secs: float = 600, stop_after: int = 0, max_retries: int = 5):
+        super().__init__(c)
+        self._stop_requested = False
+        self._records_emitted = 0
+        self._stop_after = stop_after
+        self._interval_secs = interval_secs
+        self._max_retries = max_retries
+        self._controller = RSSController(feed_input=feed_input, batch_size=c.pipeline_batch_size)
+
+    @property
+    def name(self) -> str:
+        return "from-rss"
+
+    def stop(self):
+        """
+        Stop the RSS source stage.
+        """
+        self._stop_requested = True
+        return super().stop()
+
+    def supports_cpp_node(self):
+        return False
+
+    def _fetch_feeds(self):
+        """
+        Fetch RSS feed entries and yield as MessageMeta messages.
+        """
+        retries = 0
+
+        while (not self._stop_requested) and (retries < self._max_retries):
+            try:
+                for entry_accumulator in self._controller.fetch_entries():
+                    if entry_accumulator is not None and entry_accumulator:
+                        logger.debug("Processing %d new entries...", len(entry_accumulator))
+                        df = self._controller.create_dataframe(entry_accumulator)
```

Comment:
No need to call `self._controller.create_dataframe(entry_accumulator)`. Make a method called `fetch_dataframes()` on the controller which does the conversion inline.